AmplifyPlugins/API/Sources/AWSAPIPlugin/Interceptor/RequestInterceptor/IAMURLRequestInterceptor.swift

```diff
@@ -26,52 +26,56 @@ struct IAMURLRequestInterceptor: URLRequestInterceptor {
     }
 
     func intercept(_ request: URLRequest) async throws -> URLRequest {
-        guard let mutableRequest = (request as NSURLRequest).mutableCopy() as? NSMutableURLRequest else {
-            throw APIError.unknown("Could not get mutable request", "")
-        }
-        guard let url = mutableRequest.url else {
+        var request = request
+
+        guard let url = request.url else {
             throw APIError.unknown("Could not get url from mutable request", "")
         }
         guard let host = url.host else {
             throw APIError.unknown("Could not get host from mutable request", "")
         }
 
-        mutableRequest.setValue(URLRequestConstants.ContentType.applicationJson, forHTTPHeaderField: URLRequestConstants.Header.contentType)
-        mutableRequest.setValue(host, forHTTPHeaderField: "host")
-        mutableRequest.setValue(AWSAPIPluginsCore.baseUserAgent(), forHTTPHeaderField: URLRequestConstants.Header.userAgent)
+        request.setValue(URLRequestConstants.ContentType.applicationJson, forHTTPHeaderField: URLRequestConstants.Header.contentType)
+        request.setValue(host, forHTTPHeaderField: "host")
+        request.setValue(AWSAPIPluginsCore.baseUserAgent(), forHTTPHeaderField: URLRequestConstants.Header.userAgent)
 
-        let httpMethod = HttpMethodType(rawValue: mutableRequest.httpMethod.uppercased()) ?? .get
+        let httpMethod = (request.httpMethod?.uppercased())
+            .flatMap(HttpMethodType.init(rawValue:)) ?? .get
+
```

Comment:
was this just a code preference change to chain it with `flatMap`?

Reply:
Not a preference change per se. Dropping the use of `NSMutableURLRequest` means that `httpMethod` is now optional. So it needs to be unwrapped before feeding it to `HttpMethodType` where it previously didn't. I took the one-liner approach rather than `if let else` or multiple ternary operators. I'm not opposed to moving to `if let else` if you think that's better for readability/maintainability though.